core/src/apps/cardano/helpers/address_credential_policy.py

```diff
@@ -9,90 +8,80 @@
     from ..seed import Keychain
 
 
-CREDENTIAL_POLICY_SHOW = 0
-CREDENTIAL_POLICY_SHOW_KEY_PATH = 1 << 1
-CREDENTIAL_POLICY_SHOW_CHANGE_OUTPUT = 1 << 2
-CREDENTIAL_POLICY_SHOW_ADDRESS = 1 << 3
-CREDENTIAL_POLICY_WARN_SCRIPT = 1 << 4
-CREDENTIAL_POLICY_WARN_POINTER = 1 << 5
-CREDENTIAL_POLICY_WARN_REWARD_ADDRESS = 1 << 6
-CREDENTIAL_POLICY_WARN_NO_STAKING = 1 << 7
-CREDENTIAL_POLICY_FAIL_OR_WARN_UNUSUAL = 1 << 8
-CREDENTIAL_POLICY_WARN_MISMATCH = 1 << 9
-
-CREDENTIAL_POLICIES_WARN = (
-    CREDENTIAL_POLICY_WARN_SCRIPT,
-    CREDENTIAL_POLICY_WARN_POINTER,
-    CREDENTIAL_POLICY_WARN_REWARD_ADDRESS,
-    CREDENTIAL_POLICY_WARN_NO_STAKING,
-    CREDENTIAL_POLICY_FAIL_OR_WARN_UNUSUAL,
-    CREDENTIAL_POLICY_WARN_MISMATCH,
-)
-
 ADDRESS_POLICY_SHOW_SPLIT = 0
 ADDRESS_POLICY_SHOW_SIMPLE = 1
 
 
+class CredentialPolicy:
+    # fmt: off
+
+    # use these flags for property checking
+    FLAG_WARN         = 1 << 1
+    FLAG_REWARD       = 1 << 2
+    FLAG_NO_STAKING   = 1 << 3
+    FLAG_MISMATCH     = 1 << 4
+    FLAG_UNUSUAL_PATH = 1 << 5
+
+    # these are the possible credential policies (allowed flag combinations)
+    NONE = 0
+    WARN_GENERAL               = FLAG_WARN
+    WARN_REWARD                = FLAG_WARN | FLAG_REWARD
+    WARN_NO_STAKING            = FLAG_WARN | FLAG_NO_STAKING
+    WARN_MISMATCH              = FLAG_WARN | FLAG_MISMATCH
+    WARN_UNUSUAL_PATH          = FLAG_WARN | FLAG_UNUSUAL_PATH
+    WARN_MISMATCH_UNUSUAL_PATH = FLAG_WARN | FLAG_MISMATCH | FLAG_UNUSUAL_PATH
+
+    # fmt: on
+
+    @classmethod
+    def maybe_unusual(cls, is_unusual: bool, policy: int) -> int:
+        """Helper function that optionally adds FLAG_WARN and FLAG_UNUSUAL_PATH flags."""
+        return policy | cls.WARN_UNUSUAL_PATH if is_unusual else policy
+
+
 def get_address_policy(
     keychain: Keychain, address_parameters: CardanoAddressParametersType
```

Comment:
The `keychain` parameter is now unused. Please also check callers of this function if they need the parameter.

Reply:
Hmm, I thought `make pystyle_check` would catch unused arguments. Thanks for noticing, fixed.